electrum/commands.py

```
@@ -1598,108 +1943,26 @@ def eval_bool(x: str) -> bool:
-    'pending':     (None, "Show only pending requests."),
-    'push_amount': (None, 'Push initial amount (in BTC)'),
-    'zeroconf':    (None, 'request zeroconf channel'),
-    'expired':     (None, "Show only expired requests."),
-    'paid':        (None, "Show only paid requests."),
-    'show_addresses': (None, "Show input and output addresses"),
-    'show_fiat':   (None, "Show fiat value of transactions"),
-    'show_fees':   (None, "Show miner fees paid by transactions"),
-    'year':        (None, "Show history for a given year"),
-    'from_height': (None, "Only show transactions that confirmed after given block height"),
-    'to_height':   (None, "Only show transactions that confirmed before given block height"),
-    'iknowwhatimdoing': (None, "Acknowledge that I understand the full implications of what I am about to do"),
-    'gossip':      (None, "Apply command to gossip node instead of wallet"),
-    'connection_string':      (None, "Lightning network node ID or network address"),
-    'new_fee_rate': (None, f"The Updated/Increased Transaction fee rate (in {util.UI_UNIT_NAME_FEERATE_SAT_PER_VBYTE})"),
-    'from_amount': (None, "Amount to convert (default: 1)"),
-    'from_ccy':    (None, "Currency to convert from"),
-    'to_ccy':      (None, "Currency to convert to"),
-    'public':      (None, 'Channel will be announced'),
-    'dummy_hops':  (None, 'Number of dummy hops to add'),
-}
 
 
 # don't use floats because of rounding errors
 from .transaction import convert_raw_tx_to_hex
 json_loads = lambda x: json.loads(x, parse_float=lambda x: str(to_decimal(x)))
+def check_txid(txid):
+    if not is_hash256_str(txid):
+        raise UserFacingException(f"{repr(txid)} is not a txid")
+    return txid
+
 arg_types = {
-    'num': int,
-    'nbits': int,
-    'imax': int,
-    'year': int,
-    'from_height': int,
-    'to_height': int,
+    'int': int,
+    'bool': eval_bool,
```

Could it be dangerous that `eval_bool()` always returns True for inputs not correctly representing a bool?
e.g.:
```
eval_bool("Falsee") -> True
eval_bool("Some other misplaced argument") -> True
```

---

indeed; let it raise an exception